File: 240503/되돌아오기/come-back.java
import java.util.Scanner;
public class Main {
    static int[] dx = new int[] {1,0,-1,0};//E S W N
    static int[] dy = new int[] {0,-1,0,1};
    public static int dir(char c) {
        if(c == 'E')
            return 0;
        if(c == 'S')
            return 1;
        if(c == 'W')
            return 2;
        if(c == 'N')
            return 3;
        else
            return -1;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int xpos = 0;
        int ypos = 0;
        int time = 0;
        for(int i=0;i<n;i++) {
            char c = sc.next().charAt(0);
            int dir = dir(c);
            int hop = sc.nextInt();
            for(int j=0;j<hop;j++) {
                xpos += dx[dir];
                ypos += dy[dir];
                time++;
                if (xpos == 0 && ypos == 0) {
                    System.out.println(time);
                    return;
                }
            }
        }
        if(time == n) {
            System.out.println(-1);
        }

    }
}

File: 240424/되돌아오기/come-back.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    var cnt = 0
    val n = sc.nextInt()
    var x = 0
    var y = 0

    val dx = intArrayOf(1, 0, -1, 0)
    val dy = intArrayOf(0, -1, 0, 1)

    repeat(n) {
        val direction = sc.next()[0]
        val distance = sc.nextInt()
        val dirNum = getDir(direction)

        repeat(distance) {
            cnt++

            x += dx[dirNum]
            y += dy[dirNum]

            if (x == 0 && y == 0) {
                print(cnt)
                return
            }
        }
    }

    print(-1)
}

fun getDir(direction: Char): Int {
    return when(direction) {
        'E' -> { 0 }
        'S' -> { 1 }
        'W' -> { 2 }
        else -> { 3 }
    }
}